transport/awslambda/server.go

```diff
@@ -0,0 +1,121 @@
+package awslambda
+
+import (
+	"context"
+
+	"github.com/go-kit/kit/endpoint"
+	"github.com/go-kit/kit/log"
+)
+
+// Server wraps an endpoint.
+type Server struct {
+	e            endpoint.Endpoint
+	dec          DecodeRequestFunc
+	enc          EncodeResponseFunc
+	before       []ServerRequestFunc
+	after        []ServerResponseFunc
+	errorEncoder ErrorEncoder
+	finalizer    []ServerFinalizerFunc
+	logger       log.Logger
+}
+
+// NewServer constructs a new server, which implements
+// the AWS lambda.Handler interface.
+func NewServer(
+	e endpoint.Endpoint,
+	dec DecodeRequestFunc,
+	enc EncodeResponseFunc,
+	options ...ServerOption,
+) *Server {
+	s := &Server{
+		e:      e,
+		dec:    dec,
+		enc:    enc,
+		logger: log.NewNopLogger(),
+	}
+	for _, option := range options {
+		option(s)
+	}
+	return s
+}
+
+// ServerOption sets an optional parameter for servers.
+type ServerOption func(*Server)
+
+// ServerBefore functions are executed on the payload byte,
+// before the request is decoded.
+func ServerBefore(before ...ServerRequestFunc) ServerOption {
+	return func(s *Server) { s.before = append(s.before, before...) }
+}
+
+// ServerAfter functions are only executed after invoking the endpoint
+// but prior to returning a response.
+func ServerAfter(after ...ServerResponseFunc) ServerOption {
+	return func(s *Server) { s.after = append(s.after, after...) }
+}
+
+// ServerErrorLogger is used to log non-terminal errors.
+// By default, no errors are logged.
+func ServerErrorLogger(logger log.Logger) ServerOption {
+	return func(s *Server) { s.logger = logger }
+}
+
+// ServerErrorEncoder is used to encode errors.
+func ServerErrorEncoder(ee ErrorEncoder) ServerOption {
+	return func(s *Server) { s.errorEncoder = ee }
+}
+
+// ServerFinalizer sets finalizer which are called at the end of
+// request. By default no finalizer is registered.
+func ServerFinalizer(f ...ServerFinalizerFunc) ServerOption {
+	return func(s *Server) { s.finalizer = append(s.finalizer, f...) }
+}
+
+// Invoke represents implementation of the AWS lambda.Handler interface.
+func (s *Server) Invoke(
+	ctx context.Context, payload []byte,
```

Comment:
Minor nit: if you're going to break out parameters to a new line, please keep them to one per line.

Reply:
updated 3de95b5cc04b9fde6982631c339f757bafd83ef0